index/secondary_index.cc

```diff
@@ -106,6 +106,26 @@ sstring target_parser::get_target_column_name_from_string(const sstring& targets
     return targets;
 }
 
+sstring target_parser::get_base_column_name_from_target(const sstring& target) {
+    sstring base_name = target;
+    
+    // extract column name if the target is in format "function(column_name)"
+    auto opening_pos = target.find("(");
```

Comment:
You need to *first* check if the target begins with a quote, because if it does it means this is a column name, even if it contains the "(" character (which is allowed in a column name).
I think index_target::column_name_from_target_string already does this correctly.
Please see commit https://github.com/scylladb/scylladb/pull/10841/commits/2c244c6e09c4693a47eadded930651599f8145d1 for an explanation of the different target encodings.